### ISSUE TITLE
[Accessability Bug] Self Identification dropdown Screen Reader Issues

### ISSUE THREAD

**Issue by AminaSalah** [CDH-Studio/UpSkill#1550]
**Describe the bug**
Contact Information Form:
The Self Identification dropdown/visibility selector (Private, My Circle, Public) is missing some screen reader text and some of the text read aloud is incorrect.

1. When you tab to this selector it does not state which option is currently chosen
2. When you go through the options in the selector the screen reader text that is read aloud is not accurate. 
2.a) Public is read as 1 out of 2
2.b) My Circle read as 2/3.  
2.c) Private is read as 2/3

**To Reproduce**

1. Go to Contact Information Form
2. Tab to the Self Identification selector (it does not read the current selected option)
3. Use the arrows to go through the selector (the order of the elements does not match the order read by the screen reader)

**Expected behavior**
They should all be out of 3 since there are 3 total options in the list. Also the order that the screen reader reads out doesn’t match the order that is visible when the dropdown is expanded. When you tab to this selector the screen reader states which option is currently chosen and the numbering for each option is read as follows:
- Public is read as 1 out of 3
- My Circle read as 2/3.  
- Private is read as 3/3
